src/MessageTrait.php

```diff
@@ -170,8 +178,11 @@ public function withHeader($header, $value)
             );
         }
 
+        $lowerHeader = strtolower($header);
+
         $new = clone $this;
-        $new->headers[$header] = $value;
+        $new->headerNames[$lowerHeader] = $header;
+        $new->headers[$lowerHeader] = $value;
```

Comment:
I'm a little bit lost. What should I do? :) IMO, dealing with weird header name cases will complexify a lot the base code for not real gain...

The PR in its current state keeps the last header entry used in case of "trycky case", should I update it to use the first one or we keep as it is? :)